driver-async/src/main/com/mongodb/async/client/Observer.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.async.client;
+
+/**
+ * Provides a mechanism for receiving push-based notifications.
+ *
+ * <p>
+ * Will receive a call to {@link #onSubscribe(Subscription)} on subscription to the Observable.
```

Comment:
I previously had that but had to change it, because you can also subscribe to any method that takes a `SingleResultCallback`.